sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -71,107 +35,39 @@ export class DeviceCodeCredential implements TokenCredential {
    * @param clientId - The client (application) ID of an App Registration in the tenant.
    *                   By default we will try to use the Azure CLI's client ID to authenticate.
    * @param userPromptCallback - A callback function that will be invoked to show
-                               {@link DeviceCodeInfo} to the user. If left unassigned, we will automatically log the device code information and the authentication instructions in the console.
+   *                             {@link DeviceCodeInfo} to the user. If left unassigned, we will automatically log the device code information and the authentication instructions in the console.
+   *                             Users can also pass the options as the third parameter, and skip the prompt callback entirely.
    * @param options - Options for configuring the client which makes the authentication request.
+   *                  Users can also pass the options as the third parameter, and skip the prompt callback entirely.
    */
+  constructor(tenantId?: string, clientId?: string, options?: DeviceCodeCredentialOptions);
   constructor(
-    tenantId: string = "organizations",
-    clientId: string = DeveloperSignOnClientId,
-    userPromptCallback: DeviceCodePromptCallback = defaultDeviceCodePromptCallback,
-    options?: TokenCredentialOptions
+    tenantId?: string,
+    clientId?: string,
+    userPromptCallback?: DeviceCodePromptCallback,
+    options?: DeviceCodeCredentialOptions
+  );
+  constructor(
+    tenantId?: string,
+    clientId?: string,
+    userPromptCallback?: DeviceCodePromptCallback | DeviceCodeCredentialOptions,
```

Comment:
It looks like you did this to essentially enable specifying options without specifying the callback, but I worry that this is a bit confusing and it might make it harder to interpret code written against this. Since the first two parameters are also optional as well, perhaps it would be more strait forward to allow specifying options as the first parameter, and users can set whatever combination they want in the options. Thoughts?

Reply:
Let's sync on a call!

Reply:
Just to elaborate before we call:

userPromptCallback was already optional, since we defaulted on `DeviceCodePromptCallback`. However, to use the default one AND pass optional parameters, users needed to pass `undefined`.

Since we're moving to 2.0.0, one better way to address this is to move userPromptCallback to the options bag. How does that sound?

Reply:
On our call we agreed on having just one overload to this credential by which users can pass only the options bag. I'll resolve this since I'll address it with the next things I'll push.